README.md

```diff
@@ -14,47 +23,127 @@ And then execute:
 
     $ bundle
 
-Or install it yourself as:
+Or execute the following line to install the gem without bundler.
 
     $ gem install ricohapi-mstorage
 
 ## Usage
 
 You can find [a working rails sample app using Ricoh Media Storage API](https://github.com/ricohapi/media-storage-sample-app).
 
-### Obtaining Access Token for Media Storage API
+## SDK API
 
-To obtain access tokens, you can use [ricohapi-auth gem](https://github.com/ricohapi/auth-rb).
+### Obtain an access token
+
+```ruby
+client = RicohAPI::Auth::Client.new(
+  '<your-client-id>',
+  '<your-client-secret>'
+)
+client.resource_owner_credentials = '<your-user-id>', '<your-password>'
+api_session = client.api_token_for! RicohAPI::MStorage::SCOPE
+your_access_token = api_session.access_token
+```
 
-### Media Storage API Client
+### Constructor
 
 ```ruby
 mstorage = RicohAPI::MStorage::Client.new(
-  '<your-access-token>'
+  your_access_token
 )
+```
+
+### Upload
+
+```ruby
+mstorage.upload content: file
+```
+
+### Download
 
-# GET /v1/media
+```ruby
+mstorage.download '<media-id>'
+```
+
+### List media ids
+* Without options
+You'll get a default list if you do not set any parameters.
+```ruby
 mstorage.list
+```
+
+* With options
+
+You can also use listing options. The available options are `limit`, `after` and `before`.
+```ruby
 mstorage.list limit: 25
 mstorage.list after: '<cursor-id>'
+```
+
+* Search
+
+Also, you can get a list searched by user metadata.
+```ruby
+filter = {'meta.user.<key1>'=>'<value1>', 'meta.user.<key2>'=>'<value2>'}
+mstorage.list limit: 25, after: '<cursor-id>', filter: filter
+```
 
-# GET /v1/media/{media-id}
+### Delete
+
+```ruby
+mstorage.delete '<media-id>'
+```
+
+### Get media information
+
+```ruby
 mstorage.info '<media-id>'
+```
 
-# GET /v1/media/{media-id}/meta
+### Attach media metadata
+You can define your original metadata as a 'user metadata'.
+Existing metadata value for the same key will be overwritten. Up to 10 user metadata can be attached to a media data.
+
+```ruby
+mstorage.addMeta '<media-id>', {'user.<key1>'=>'<value1>', 'user.<key2>'=>'<value2>'}
+```
+
+### Get media metadata
+* All
+```ruby
 mstorage.meta '<media-id>'
+```
 
-# GET /v1/media/{media-id}/content
-mstorage.download '<media-id>'
+* Exif
+```ruby
+mstorage.meta '<media-id>', 'exif'
+```
 
-# POST /v1/media (multipart)
-mstorage.upload content: file
+* Google Photo Sphere XMP
+```ruby
+mstorage.meta '<media-id>', 'gpano'
+```
 
-# DELETE /v1/media/{media-id}
-mstorage.delete '<media-id>'
+* User metadata (all)
+```ruby
+mstorage.meta '<media-id>', 'user'
 ```
 
-## License
+* User metadata (with a key)
+```ruby
+mstorage.meta '<media-id>', 'user.<key>'
+```
 
-The gem is available as open source under the terms of the [MIT License](http://opensource.org/licenses/MIT).
+### Delete media metadata
+* User metadata (all)
+```ruby
+mstorage.removeMeta '<media-id>', 'user'
```

Comment:
You should `mstorage.remove_meta ....` .
You should use the snake case for method name.